pkg/services/datasources/models.go

```diff
@@ -150,60 +150,54 @@ func (e ErrDatasourceSecretsPluginUserFriendly) Error() string {
 // ----------------------
 // COMMANDS
 
-// Also acts as api DTO
-type AddDataSourceCommand struct {
+// The input values shared with both Add+Update commands (also part of the DTO)
+// This part of the command can share validation between add and update actions
+type BaseDataSourceCommand struct {
 	Name            string            `json:"name"`
 	Type            string            `json:"type" binding:"Required"`
 	Access          DsAccess          `json:"access" binding:"Required"`
 	URL             string            `json:"url"`
-	Database        string            `json:"database"`
 	User            string            `json:"user"`
+	Database        string            `json:"database"`
 	BasicAuth       bool              `json:"basicAuth"`
 	BasicAuthUser   string            `json:"basicAuthUser"`
 	WithCredentials bool              `json:"withCredentials"`
 	IsDefault       bool              `json:"isDefault"`
 	JsonData        *simplejson.Json  `json:"jsonData"`
 	SecureJsonData  map[string]string `json:"secureJsonData"`
 	UID             string            `json:"uid"`
+
 	// swagger:ignore
-	APIVersion string `json:"apiVersion"`
+	APIVersion string `json:"apiVersion,omitempty"`
 	// swagger:ignore
 	IsPrunable bool
 
 	OrgID                   int64             `json:"-"`
-	UserID                  int64             `json:"-"`
 	ReadOnly                bool              `json:"-"`
 	EncryptedSecureJsonData map[string][]byte `json:"-"`
 	UpdateSecretFn          UpdateSecretFn    `json:"-"`
 }
 
+// Also acts as api DTO
+type AddDataSourceCommand struct {
+	BaseDataSourceCommand
```

Comment:
Nit. Does this need json inline tag to make the JSON unmarshal work as expected?